OpenRA.Mods.Common/Activities/UnloadCargo.cs

```diff
@@ -56,40 +59,42 @@ IEnumerable<CPos> BlockedExitCells(Actor passenger)
 		public override Activity Tick(Actor self)
 		{
 			cargo.Unloading = false;
-			if (IsCanceled || cargo.IsEmpty(self))
+			if (IsCanceled || !actorsToUnload.Any() || !cargo.Passengers.Intersect(actorsToUnload).Any())
```

Comment:
I'd suggest `!actorsToUnload.Any(x => cargo.Passangers.Contains(x))` here, mostly because it's shorter (these collections don't really have enough elements to make a difference regarding performance).